### ISSUE TITLE
Can't install the plugin

### ISSUE THREAD

**Issue by mikigp** [lionep/homebridge-own#3]
I tried to install the plugin... but after a good start, the process stops. Here is the log:
```
pi@raspberrypi:~ $ sudo npm -g install homebridge-own

> mdns@2.3.3 install /usr/lib/node_modules/homebridge-own/node_modules/mdns
> node-gyp rebuild

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/7.3.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/homebridge-own/node_modules/mdns/.node-gyp"
make: Entering directory '/usr/lib/node_modules/homebridge-own/node_modules/mdns/build'
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_sd.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_browse.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_enumerate_domains.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_get_addr_info.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_process_result.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref_deallocate.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref_sock_fd.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_register.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_resolve.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_update_record.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/mdns_utils.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/network_interface.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/socket_watcher.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_ref.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_create.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_deallocate.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_set_value.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_get_length.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_buffer_to_object.o
  SOLINK_MODULE(target) Release/obj.target/dns_sd_bindings.node
  COPY Release/dns_sd_bindings.node
make: Leaving directory '/usr/lib/node_modules/homebridge-own/node_modules/mdns/build'

> curve25519@1.1.0 install /usr/lib/node_modules/homebridge-own/node_modules/curve25519
> node-gyp rebuild

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/7.3.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/homebridge-own/node_modules/curve25519/.node-gyp"
make: Entering directory '/usr/lib/node_modules/homebridge-own/node_modules/curve25519/build'
  CXX(target) Release/obj.target/curve/node_curve.o
  SOLINK_MODULE(target) Release/obj.target/curve.node
  COPY Release/curve.node
make: Leaving directory '/usr/lib/node_modules/homebridge-own/node_modules/curve25519/build'

> ed25519@0.0.3 install /usr/lib/node_modules/homebridge-own/node_modules/ed25519
> node-gyp rebuild

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/7.3.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/homebridge-own/node_modules/ed25519/.node-gyp"
make: Entering directory '/usr/lib/node_modules/homebridge-own/node_modules/ed25519/build'
  CC(target) Release/obj.target/ed25519/src/ed25519/keypair.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sign.o
  CC(target) Release/obj.target/ed25519/src/ed25519/open.o
  CC(target) Release/obj.target/ed25519/src/ed25519/crypto_verify_32.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_double_scalarmult.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_scalarmult_base.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_precomp_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_cached.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_madd.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_msub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p3.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_1.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_cmov.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_copy.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_neg.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_mul.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_invert.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnegative.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnonzero.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_pow22523.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_reduce.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_muladd.o
  CXX(target) Release/obj.target/ed25519/src/ed25519.o
  SOLINK_MODULE(target) Release/obj.target/ed25519.node
  COPY Release/ed25519.node
make: Leaving directory '/usr/lib/node_modules/homebridge-own/node_modules/ed25519/build'

> bignum@0.12.5 install /usr/lib/node_modules/homebridge-own/node_modules/bignum
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(403): https://rvagg-node.s3-us-west-2.amazonaws.com/bignum/v0.12.5/bignum-v0.12.5-node-v51-linux-arm.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for bignum@0.12.5 and node@7.3.0 (node-v51 ABI) (falling back to source compile with node-gyp) 
gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/7.3.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/homebridge-own/node_modules/bignum/.node-gyp"
make: Entering directory '/usr/lib/node_modules/homebridge-own/node_modules/bignum/build'
make: *** No rule to make target '../.node-gyp/7.3.0/include/node/common.gypi', needed by 'Makefile'.  Stop.
make: Leaving directory '/usr/lib/node_modules/homebridge-own/node_modules/bignum/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 4.4.38-v7+
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding/bignum.node" "--module_name=bignum" "--module_path=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding"
gyp ERR! cwd /usr/lib/node_modules/homebridge-own/node_modules/bignum
gyp ERR! node -v v7.3.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding/bignum.node --module_name=bignum --module_path=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/homebridge-own/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:106:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:885:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
node-pre-gyp ERR! System Linux 4.4.38-v7+
node-pre-gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/homebridge-own/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/lib/node_modules/homebridge-own/node_modules/bignum
node-pre-gyp ERR! node -v v7.3.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.32
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding/bignum.node --module_name=bignum --module_path=/usr/lib/node_modules/homebridge-own/node_modules/bignum/binding' (1)
/usr/lib
└── (empty)

npm ERR! Linux 4.4.38-v7+
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "-g" "install" "homebridge-own"
npm ERR! node v7.3.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! bignum@0.12.5 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bignum@0.12.5 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bignum package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bignum
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bignum
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pi/npm-debug.log
npm ERR! code 1
```
Can anyone help me?

**Comment by lionep** [comment]
Try with the `--unsafe-perm` flag as explained [here](https://github.com/nfarina/homebridge#installation)

**Comment by mikigp** [comment]
just the plugin or I have to reinstall homebridge?

**Comment by lionep** [comment]
I guess just for the plugin.

**Comment by mikigp** [comment]
Solved installation as you suggested: `sudo npm install -g --unsafe-perm homebridge-own`
Started homebridge: the problem now is that I can't see any extra device. (the plugin is loaded correctly and the config.json as been modified as instructions...) Tried to change port number... no luck!

**Comment by lionep** [comment]
Are you sure about the MyHome server ip ? by default it's setup to 192.168.0.20, but it might be different on your network.

**Comment by mikigp** [comment]
I'm sure: I normally use the http page to interact with my system...

**Comment by mikigp** [comment]
This is the homebridge starting log:
```
[1/4/2017, 6:54:28 PM] Loading 3 platforms...
[...]
[1/4/2017, 6:54:28 PM] [MyHomeTest] Initializing own platform...
[1/4/2017, 6:54:28 PM] [MyHomeTest] Constructing platform...
[1/4/2017, 6:54:28 PM] [MyHomeTest] Fetch OWN accessories
[1/4/2017, 6:54:28 PM] Loading 3 accessories...
[...]
[1/4/2017, 6:54:28 PM] [MyHomeTest] Lights : []
[1/4/2017, 6:54:28 PM] [MyHomeTest] Automations : []
[1/4/2017, 6:54:28 PM] [MyHomeTest] Temp sensors : []
```

When I tried to restart MyHome, homebridge failed the initialization... so I think that the connection is ok.. just no accessories are loaded... right?

**Comment by lionep** [comment]
What's the model of your server ?
Someone had issues with one Bticino F455, which does not behave correctly.

**Comment by mikigp** [comment]
Model: F454
FW: 2.0.46
HW: 0.0.0
Micro Version: 14.49.02
Kernel: 2.6.32

**Comment by mikigp** [comment]
Any update?

**Comment by lionep** [comment]
Can you try this command on your network 
```
telnet {server_ip} 20000
```

Then, server should respond `*#*1##`
Type this command `*#1*0##` and press enter.

What's the result ?

**Comment by mikigp** [comment]
Here is the result (I tried two times with the telnet command):
```
Trying 192.168.0.210...
Connected to 192.168.0.210.
Escape character is '^]'.
*#*1##*#1*0##
Connection closed by foreign host.Trying 192.168.0.210...
Connected to 192.168.0.210.
Escape character is '^]'.
*#*1##*#1*0##
Connection closed by foreign host.
```

**Comment by lionep** [comment]
Should be solved by release `v0.3.1`
